Index: git/src/network/networkd-link.c
===================================================================
--- git.orig/src/network/networkd-link.c
+++ git/src/network/networkd-link.c
@@ -2040,9 +2040,14 @@ static int link_configure(Link *link) {
         assert(link->network);
         assert(link->state == LINK_STATE_PENDING);
 
-        r = link_drop_foreign_config(link);
-        if (r < 0)
-                return r;
+        /* Drop foreign config, but ignore loopback or critical devices.  We
+         * do not want to remove loopback address or addresses used for root
+         * NFS. */
+        if (!(link->flags & IFF_LOOPBACK) && !(link->network->dhcp_critical)) {
+                r = link_drop_foreign_config(link);
+                if (r < 0)
+                        return r;
+        }
 
         r = link_set_bridge_fdb(link);
         if (r < 0)
