Index: git/units/sysinit.target
===================================================================
--- git.orig/units/sysinit.target
+++ git/units/sysinit.target
@@ -10,6 +10,6 @@
 [Unit]
 Description=System Initialization
 Conflicts=emergency.service emergency.target
-Wants=local-fs.target swap.target
+Wants=local-fs.target
 After=local-fs.target swap.target emergency.service emergency.target
 RefuseManualStart=yes
