Index: git/src/libsystemd/sd-rtnl/rtnl-types.c
===================================================================
--- git.orig/src/libsystemd/sd-rtnl/rtnl-types.c
+++ git/src/libsystemd/sd-rtnl/rtnl-types.c
@@ -39,6 +39,50 @@
 #include "rtnl-types.h"
 #include "missing.h"
 
+#ifndef IFLA_BRIDGE_MAX
+#  define NO_KERNEL_3_10	1
+
+#  define IFLA_BRIDGE_MAX       2
+#  define IFLA_BRIDGE_FLAGS     0
+#  define IFLA_BRIDGE_MODE	1
+#endif
+
+#ifndef IFLA_IPTUN_MAX
+enum {
+        IFLA_IPTUN_UNSPEC,
+        IFLA_IPTUN_LINK,
+        IFLA_IPTUN_LOCAL,
+        IFLA_IPTUN_REMOTE,
+        IFLA_IPTUN_TTL,
+        IFLA_IPTUN_TOS,
+        IFLA_IPTUN_ENCAP_LIMIT,
+        IFLA_IPTUN_FLOWINFO,
+        IFLA_IPTUN_FLAGS,
+        IFLA_IPTUN_PROTO,
+        IFLA_IPTUN_PMTUDISC,
+        IFLA_IPTUN_6RD_PREFIX,
+        IFLA_IPTUN_6RD_RELAY_PREFIX,
+        IFLA_IPTUN_6RD_PREFIXLEN,
+        IFLA_IPTUN_6RD_RELAY_PREFIXLEN,
+        __IFLA_IPTUN_MAX,
+};
+#define IFLA_IPTUN_MAX  (__IFLA_IPTUN_MAX - 1)
+#endif
+
+#ifndef IFLA_VTI_MAX
+enum {
+        IFLA_VTI_UNSPEC,
+        IFLA_VTI_LINK,
+        IFLA_VTI_IKEY,
+        IFLA_VTI_OKEY,
+        IFLA_VTI_LOCAL,
+        IFLA_VTI_REMOTE,
+        __IFLA_VTI_MAX,
+};
+
+#define IFLA_VTI_MAX    (__IFLA_VTI_MAX - 1)
+#endif
+
 static const NLTypeSystem rtnl_link_type_system;
 
 static const NLType rtnl_link_info_data_veth_types[VETH_INFO_MAX + 1] = {
@@ -48,7 +92,9 @@ static const NLType rtnl_link_info_data_
 
 static const NLType rtnl_link_info_data_macvlan_types[IFLA_MACVLAN_MAX + 1] = {
         [IFLA_MACVLAN_MODE]  = { .type = NLA_U32 },
+#ifndef NO_KERNEL_3_10
         [IFLA_MACVLAN_FLAGS] = { .type = NLA_U16 },
+#endif
 };
 
 static const NLType rtnl_link_info_data_bridge_types[IFLA_BRIDGE_MAX + 1] = {
@@ -67,9 +113,9 @@ static const NLType rtnl_link_info_data_
         [IFLA_VLAN_EGRESS_QOS]  = { .type = NLA_NESTED },
         [IFLA_VLAN_INGRESS_QOS] = { .type = NLA_NESTED },
 */
-#ifdef IFLA_VLAN_PROTOCOL
+#ifndef IFLA_BRIDGE_MAX
         [IFLA_VLAN_PROTOCOL]    = { .type = NLA_U16 },
-#endif
+#endif
 };
 
 static const NLType rtnl_link_info_data_bond_types[IFLA_BOND_MAX + 1] = {
@@ -253,6 +301,7 @@ static const NLType rtnl_link_types[IFLA
 */
         [IFLA_GROUP]            = { .type = NLA_U32 },
         [IFLA_NET_NS_FD]        = { .type = NLA_U32 },
+#ifndef NO_KERNEL_3_10
         [IFLA_EXT_MASK]         = { .type = NLA_U32 },
         [IFLA_PROMISCUITY]      = { .type = NLA_U32 },
         [IFLA_NUM_TX_QUEUES]    = { .type = NLA_U32 },
@@ -261,6 +310,7 @@ static const NLType rtnl_link_types[IFLA
 /*
         [IFLA_PHYS_PORT_ID]     = { .type = NLA_BINARY, .len = MAX_PHYS_PORT_ID_LEN },
 */
+#endif
 };
 
 static const NLTypeSystem rtnl_link_type_system = {
Index: git/src/test/test-rtnl-manual.c
===================================================================
--- git.orig/src/test/test-rtnl-manual.c
+++ git/src/test/test-rtnl-manual.c
@@ -34,6 +34,28 @@
 #include "event-util.h"
 #include "rtnl-internal.h"
 
+#ifndef IFLA_IPTUN_MAX
+enum {
+        IFLA_IPTUN_UNSPEC,
+        IFLA_IPTUN_LINK,
+        IFLA_IPTUN_LOCAL,
+        IFLA_IPTUN_REMOTE,
+        IFLA_IPTUN_TTL,
+        IFLA_IPTUN_TOS,
+        IFLA_IPTUN_ENCAP_LIMIT,
+        IFLA_IPTUN_FLOWINFO,
+        IFLA_IPTUN_FLAGS,
+        IFLA_IPTUN_PROTO,
+        IFLA_IPTUN_PMTUDISC,
+        IFLA_IPTUN_6RD_PREFIX,
+        IFLA_IPTUN_6RD_RELAY_PREFIX,
+        IFLA_IPTUN_6RD_PREFIXLEN,
+        IFLA_IPTUN_6RD_RELAY_PREFIXLEN,
+        __IFLA_IPTUN_MAX,
+};
+#define IFLA_IPTUN_MAX  (__IFLA_IPTUN_MAX - 1)
+#endif
+
 static int load_module(const char *mod_name) {
         struct kmod_ctx *ctx;
         struct kmod_list *list = NULL, *l;
