Index: systemd-199/units/systemd-readahead-collect.service.in
===================================================================
--- systemd-199.orig/units/systemd-readahead-collect.service.in
+++ systemd-199/units/systemd-readahead-collect.service.in
@@ -12,6 +12,7 @@ DefaultDependencies=no
 Wants=systemd-readahead-done.timer
 Conflicts=shutdown.target
 Before=sysinit.target shutdown.target
+ConditionPathExists=!/run/nfs-ro-boot
 ConditionPathExists=!/run/systemd/readahead/cancel
 ConditionPathExists=!/run/systemd/readahead/done
 ConditionVirtualization=no
