Index: git/arm/start.S
===================================================================
--- git.orig/arm/start.S
+++ git/arm/start.S
@@ -25,6 +25,14 @@ _start:
 	add	a3, a2, a1, lsl #2	@ &argv[argc]
 	add	a3, a3, #4		@ envp	
 	str	a3, [ip, #0]		@ environ = envp
+
+#ifdef WANT_ELFINFO
+1:	ldr	r5, [a3], #4		@ load *envp and increment it
+	cmp	r5, #0			@ read value==0?
+	bne	1b
+	str	a3, [ip, #4]		@ __elfinfo = envp
+#endif
+
 	bl	main
 		
 @
