Index: xine-lib-1.2/src/combined/ffmpeg/ff_audio_decoder.c
===================================================================
--- xine-lib-1.2.orig/src/combined/ffmpeg/ff_audio_decoder.c
+++ xine-lib-1.2/src/combined/ffmpeg/ff_audio_decoder.c
@@ -284,12 +284,17 @@ static void ff_audio_decode_data (audio_
 
     if (!this->output_open) {
       if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels) {
+	AVPacket avpkt;
+
+	av_init_packet(&avpkt);
+	avpkt.data = &this->buf[0];
+	avpkt.size = this->size;
+
         decode_buffer_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
-        avcodec_decode_audio2 (this->context,
+        avcodec_decode_audio3 (this->context,
                               (int16_t *)this->decode_buffer,
                               &decode_buffer_size,
-                              &this->buf[0],
-                              this->size);
+			      &avpkt);
 	this->audio_bits = this->context->bits_per_sample;
 	this->audio_sample_rate = this->context->sample_rate;
 	this->audio_channels = this->context->channels;
@@ -309,12 +314,17 @@ static void ff_audio_decode_data (audio_
 
       offset = 0;
       while (this->size>0) {
+	AVPacket avpkt;
+
+	av_init_packet(&avpkt);
+	avpkt.data = &this->buf[offset];
+	avpkt.size = this->size;
+
         decode_buffer_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
-        bytes_consumed = avcodec_decode_audio2 (this->context,
+        bytes_consumed = avcodec_decode_audio3 (this->context,
                                                (int16_t *)this->decode_buffer,
                                                &decode_buffer_size,
-                                               &this->buf[offset],
-                                               this->size);
+                                               &avpkt);
 
         if (bytes_consumed<0) {
           xprintf (this->stream->xine, XINE_VERBOSITY_DEBUG,
Index: xine-lib-1.2/src/combined/ffmpeg/ff_video_decoder.c
===================================================================
--- xine-lib-1.2.orig/src/combined/ffmpeg/ff_video_decoder.c
+++ xine-lib-1.2/src/combined/ffmpeg/ff_video_decoder.c
@@ -1011,7 +1011,7 @@ static void ff_handle_mpeg12_buffer (ff_
   lprintf("handle_mpeg12_buffer\n");
 
   while ((size > 0) || (flush == 1)) {
-
+    AVPacket avpkt;
     uint8_t *current;
     int next_flush;
 
@@ -1043,16 +1043,19 @@ static void ff_handle_mpeg12_buffer (ff_
     }
 
     /* skip decoding b frames if too late */
-    this->context->hurry_up = (this->skipframes > 0);
+    this->context->skip_frame = (this->skipframes > 0) ? AVDISCARD_BIDIR : AVDISCARD_DEFAULT;
+
+    av_init_packet(&avpkt);
+    avpkt.data = this->mpeg_parser->chunk_buffer;
+    avpkt.size = this->mpeg_parser->buffer_size;
 
     lprintf("avcodec_decode_video: size=%d\n", this->mpeg_parser->buffer_size);
-    len = avcodec_decode_video (this->context, this->av_frame,
-                                &got_picture, this->mpeg_parser->chunk_buffer,
-                                this->mpeg_parser->buffer_size);
-    lprintf("avcodec_decode_video: decoded_size=%d, got_picture=%d\n",
+    len = avcodec_decode_video2 (this->context, this->av_frame,
+                                &got_picture, &avpkt);
+    lprintf("avcodec_decode_video2: decoded_size=%d, got_picture=%d\n",
             len, got_picture);
     len = current - buf->content - offset;
-    lprintf("avcodec_decode_video: consumed_size=%d\n", len);
+    lprintf("avcodec_decode_video2: consumed_size=%d\n", len);
 
     flush = next_flush;
 
@@ -1100,7 +1103,7 @@ static void ff_handle_mpeg12_buffer (ff_
 
     } else {
 
-      if (this->context->hurry_up) {
+      if (this->context->skip_frame > 0) {
         /* skipped frame, output a bad frame */
         img = this->stream->video_out->get_frame (this->stream->video_out,
                                                   this->bih.biWidth,
@@ -1287,13 +1290,18 @@ static void ff_handle_buffer (ff_video_d
         this->size = 0;
         got_picture = 0;
       } else {
+	AVPacket	avpkt;
+
+	av_init_packet(&avpkt);
+	avpkt.data = &chunk_buf[offset];
+	avpkt.size = this->size;
+
         /* skip decoding b frames if too late */
-        this->context->hurry_up = (this->skipframes > 0);
+        this->context->skip_frame = (this->skipframes > 0) ? AVDISCARD_BIDIR : AVDISCARD_DEFAULT;
 
         lprintf("buffer size: %d\n", this->size);
-        len = avcodec_decode_video (this->context, this->av_frame,
-                                    &got_picture, &chunk_buf[offset],
-                                    this->size);
+        len = avcodec_decode_video2 (this->context, this->av_frame,
+				     &got_picture, &avpkt);
 
         /* reset consumed pts value */
         this->context->reordered_opaque = ff_tag_pts(this, 0);
