Index: udev-172/udev/udevd.c
===================================================================
--- udev-172.orig/udev/udevd.c
+++ udev-172/udev/udevd.c
@@ -1180,10 +1180,11 @@ int main(int argc, char *argv[])
 	struct udev *udev;
 	FILE *f;
 	sigset_t mask;
-	int daemonize = false;
+	enum { DAEMON_FG, DAEMON_BG, DAEMON_STOP } daemonize = DAEMON_FG;
 	int resolve_names = 1;
 	static const struct option options[] = {
 		{ "daemon", no_argument, NULL, 'd' },
+		{ "daemon-stop", no_argument, NULL, 'S' },
 		{ "debug", no_argument, NULL, 'D' },
 		{ "children-max", required_argument, NULL, 'c' },
 		{ "exec-delay", required_argument, NULL, 'e' },
@@ -1236,13 +1237,16 @@ int main(int argc, char *argv[])
 	for (;;) {
 		int option;
 
-		option = getopt_long(argc, argv, "c:deDtN:hV", options, NULL);
+		option = getopt_long(argc, argv, "c:deDtN:hVS", options, NULL);
 		if (option == -1)
 			break;
 
 		switch (option) {
 		case 'd':
-			daemonize = true;
+			daemonize = DAEMON_BG;
+			break;
+		case 'S':
+			daemonize = DAEMON_STOP;
 			break;
 		case 'c':
 			children_max = strtoul(optarg, NULL, 0);
@@ -1412,7 +1416,9 @@ int main(int argc, char *argv[])
 		goto exit;
 	}
 
-	if (daemonize) {
+	if (daemonize == DAEMON_STOP)
+		raise(SIGSTOP);
+	else if (daemonize == DAEMON_BG) {
 		pid_t pid;
 		int fd;
 
