Index: setup.py
===================================================================
--- setup.py	(Revision 1118)
+++ setup.py	(Arbeitskopie)
@@ -21,8 +21,8 @@
 from distutils.core import setup
 import os, sys
 
-bbdir = os.path.join(sys.prefix, 'share', 'bitbake')
-docdir = os.path.join(sys.prefix, 'share', 'doc')
+bbdir = os.path.join('share', 'bitbake')
+docdir = os.path.join('share', 'doc')
 # bbdir = os.path.join('bitbake')
 # docdir = os.path.join('doc')
 
