Index: lib/bb/cache.py
===================================================================
--- lib/bb/cache.py.orig
+++ lib/bb/cache.py
@@ -422,9 +422,16 @@ class Cache:
         rrecommends = self.getVar('RRECOMMENDS', file_name, True) or ""
         for package in packages + [pn]:
             if not package in cacheData.rundeps[file_name]:
-                cacheData.rundeps[file_name][package] = []
+                cacheData.rundeps[file_name][package] = ""
             if not package in cacheData.runrecs[file_name]:
-                cacheData.runrecs[file_name][package] = []
+                cacheData.runrecs[file_name][package] = ""
+
+            if eval(self.getVar('WEAK_RUNTIME_DEPENDENCIES_%s' % package,
+                                file_name, True) or "False"):
+                bb.msg.debug(1, bb.msg.domain.Cache,
+                             "Ignoring weak runtime dependencies for '%s'" %
+                             package)
+                continue
 
             cacheData.rundeps[file_name][package] = rdepends + " " + (self.getVar("RDEPENDS_%s" % package, file_name, True) or "")
             cacheData.runrecs[file_name][package] = rrecommends + " " + (self.getVar("RRECOMMENDS_%s" % package, file_name, True) or "")
