Index: upstart-0.5.0/nih/Makefile.am
===================================================================
--- upstart-0.5.0.orig/nih/Makefile.am
+++ upstart-0.5.0/nih/Makefile.am
@@ -49,7 +49,7 @@ libnih_dbus_la_SOURCES = \
 	dbus.c
 
 libnih_dbus_la_CFLAGS = $(DBUS_CFLAGS)
-libnih_dbus_la_LIBS = $(DBUS_LIBS)
+libnih_dbus_la_LIBADD = $(DBUS_LIBS) libnih.la
 
 if INSTALL_NIH
 libnih_dbus_la_LDFLAGS = \
