From 9d730108cb77a439992531cf5b0c9f6fc1f00bb7 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
Date: Tue, 14 Feb 2012 16:45:25 +0100
Subject: [PATCH] start ntp service when interface is ready

---
 src/service.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/service.c b/src/service.c
index 1f99b34..e7827ea 100644
--- a/src/service.c
+++ b/src/service.c
@@ -3972,7 +3972,8 @@ static int service_indicate_state(struct connman_service *service)
 		}
 	}
 
-	if (new_state == CONNMAN_SERVICE_STATE_ONLINE)
+	if (new_state == CONNMAN_SERVICE_STATE_ONLINE ||
+	    new_state == CONNMAN_SERVICE_STATE_READY)
 		__connman_timeserver_sync(service);
 
 	if (new_state == CONNMAN_SERVICE_STATE_IDLE) {
-- 
1.7.7.6

