Index: git/src/v4l2subdev.c
===================================================================
--- git.orig/src/v4l2subdev.c
+++ git/src/v4l2subdev.c
@@ -581,6 +581,7 @@ static struct {
 	const char *name;
 	enum v4l2_mbus_pixelcode code;
 } mbus_formats[] = {
+	{ "FIXED", V4L2_MBUS_FMT_FIXED},
 	{ "Y8", V4L2_MBUS_FMT_Y8_1X8},
 	{ "Y10", V4L2_MBUS_FMT_Y10_1X10 },
 	{ "Y12", V4L2_MBUS_FMT_Y12_1X12 },
@@ -590,6 +591,12 @@ static struct {
 	{ "UYVY", V4L2_MBUS_FMT_UYVY8_1X16 },
 	{ "UYVY1_5X8", V4L2_MBUS_FMT_UYVY8_1_5X8 },
 	{ "UYVY2X8", V4L2_MBUS_FMT_UYVY8_2X8 },
+#ifdef HAVE_V4L2_MBUS_FMT_UYVY10_1X20
+	{ "UYVY10_1X20", V4L2_MBUS_FMT_UYVY10_1X20 },
+	{ "VYUY10_1X20", V4L2_MBUS_FMT_VYUY10_1X20 },
+#endif
+	{ "YUYV10_1X20", V4L2_MBUS_FMT_YUYV10_1X20 },
+	{ "YVYU10_1X20", V4L2_MBUS_FMT_YVYU10_1X20 },
 	{ "SBGGR8", V4L2_MBUS_FMT_SBGGR8_1X8 },
 	{ "SGBRG8", V4L2_MBUS_FMT_SGBRG8_1X8 },
 	{ "SGRBG8", V4L2_MBUS_FMT_SGRBG8_1X8 },
