Index: systemd-215/Makefile.am
===================================================================
--- systemd-215.orig/Makefile.am
+++ systemd-215/Makefile.am
@@ -2943,18 +2943,7 @@ libudev_core_la_SOURCES = \
 	src/udev/udev-rules.c \
 	src/udev/udev-ctrl.c \
 	src/udev/udev-builtin.c \
-	src/udev/udev-builtin-btrfs.c \
-	src/udev/udev-builtin-hwdb.c \
-	src/udev/udev-builtin-input_id.c \
-	src/udev/udev-builtin-keyboard.c \
-	src/udev/udev-builtin-net_id.c \
-	src/udev/udev-builtin-net_setup_link.c \
-	src/udev/udev-builtin-path_id.c \
-	src/udev/udev-builtin-usb_id.c \
-	src/udev/net/link-config.h \
-	src/udev/net/link-config.c \
-	src/udev/net/ethtool-util.h \
-	src/udev/net/ethtool-util.c
+	src/udev/udev-builtin-hwdb.c
 
 nodist_libudev_core_la_SOURCES = \
 	src/udev/keyboard-keys-from-name.h \
Index: systemd-215/src/udev/udev-builtin.c
===================================================================
--- systemd-215.orig/src/udev/udev-builtin.c
+++ systemd-215/src/udev/udev-builtin.c
@@ -30,26 +30,7 @@
 static bool initialized;
 
 static const struct udev_builtin *builtins[] = {
-#ifdef HAVE_BLKID
-        [UDEV_BUILTIN_BLKID] = &udev_builtin_blkid,
-#endif
-        [UDEV_BUILTIN_BTRFS] = &udev_builtin_btrfs,
-#ifdef HAVE_FIRMWARE
-        [UDEV_BUILTIN_FIRMWARE] = &udev_builtin_firmware,
-#endif
         [UDEV_BUILTIN_HWDB] = &udev_builtin_hwdb,
-        [UDEV_BUILTIN_INPUT_ID] = &udev_builtin_input_id,
-        [UDEV_BUILTIN_KEYBOARD] = &udev_builtin_keyboard,
-#ifdef HAVE_KMOD
-        [UDEV_BUILTIN_KMOD] = &udev_builtin_kmod,
-#endif
-        [UDEV_BUILTIN_NET_ID] = &udev_builtin_net_id,
-        [UDEV_BUILTIN_NET_LINK] = &udev_builtin_net_setup_link,
-        [UDEV_BUILTIN_PATH_ID] = &udev_builtin_path_id,
-        [UDEV_BUILTIN_USB_ID] = &udev_builtin_usb_id,
-#ifdef HAVE_ACL
-        [UDEV_BUILTIN_UACCESS] = &udev_builtin_uaccess,
-#endif
 };
 
 void udev_builtin_init(struct udev *udev)
Index: systemd-215/src/udev/udevadm.c
===================================================================
--- systemd-215.orig/src/udev/udevadm.c
+++ systemd-215/src/udev/udevadm.c
@@ -51,11 +51,6 @@ static const struct udevadm_cmd udevadm_
 };
 
 static const struct udevadm_cmd *udevadm_cmds[] = {
-        &udevadm_info,
-        &udevadm_trigger,
-        &udevadm_settle,
-        &udevadm_control,
-        &udevadm_monitor,
         &udevadm_hwdb,
         &udevadm_test,
         &udevadm_test_builtin,
