Index: lib/bb/cooker.py
===================================================================
--- lib/bb/cooker.py.orig
+++ lib/bb/cooker.py
@@ -592,9 +592,6 @@ class BBCooker:
         build.
         """
 
-        # Wipe the OS environment
-        bb.utils.empty_environment()
-
         if self.configuration.show_environment:
             self.showEnvironment(self.configuration.buildfile, self.configuration.pkgs_to_build)
             sys.exit( 0 )
