Index: lib/bb/cooker.py
===================================================================
--- lib/bb/cooker.py.orig
+++ lib/bb/cooker.py
@@ -923,8 +923,10 @@ class BBCooker:
                 newfiles.update(globbed)
 
         bbmask = bb.data.getVar('BBMASK', self.configuration.data, 1)
+        bbmask_basenames = (bb.data.getVar('BBMASK_BASENAMES',
+                                           self.configuration.data, 1) or "").split()
 
-        if not bbmask:
+        if not bbmask and not bbmask_basenames:
             return (list(newfiles), 0)
 
         try:
@@ -932,9 +934,17 @@ class BBCooker:
         except sre_constants.error:
             bb.msg.fatal(bb.msg.domain.Collection, "BBMASK is not a valid regular expression.")
 
+        bbmask_dict = {}
+        for f in bbmask_basenames:
+            if not f.endswith('.bb'):
+                f += '.bb'
+
+            bbmask_dict[f] = 1
+
         finalfiles = []
         for f in newfiles:
-            if bbmask_compiled.search(f):
+            f_base = os.path.basename(f)
+            if bbmask_dict.has_key(f_base) or bbmask_compiled.search(f):
                 bb.msg.debug(1, bb.msg.domain.Collection, "skipping masked file %s" % f)
                 masked += 1
                 continue
