Index: lib/bb/utils.py
===================================================================
--- lib/bb/utils.py.orig
+++ lib/bb/utils.py
@@ -181,6 +181,10 @@ def better_exec(code, context, text, rea
     error.
     """
     import bb,sys
+
+    if sys.hexversion < 0x2050000:
+        context['any'] = lambda x: len(filter(lambda e: bool(e), x)) > 0
+
     try:
         exec code in context
     except:
