Index: schroedinger-1.0.10/orc.mak
===================================================================
--- schroedinger-1.0.10.orig/orc.mak
+++ schroedinger-1.0.10/orc.mak
@@ -37,6 +37,7 @@ cp_v_gen = 
 if HAVE_ORCC
 tmp-orc.c: $(srcdir)/$(ORC_SOURCE).orc
 	$(orcc_v_gen)$(ORCC) $(ORCC_FLAGS) --inline --implementation -o tmp-orc.c $(srcdir)/$(ORC_SOURCE).orc
+	sed -i -e '11!a\' -e '#include <stdint.h>' $@
 
 $(ORC_SOURCE).h: $(srcdir)/$(ORC_SOURCE).orc
 	$(orcc_v_gen)$(ORCC) $(ORCC_FLAGS) --inline --header -o $(ORC_SOURCE).h $(srcdir)/$(ORC_SOURCE).orc
Index: schroedinger-1.0.10/schroedinger/schroorc-dist.c
===================================================================
--- schroedinger-1.0.10.orig/schroedinger/schroorc-dist.c
+++ schroedinger-1.0.10/schroedinger/schroorc-dist.c
@@ -8,6 +8,8 @@
 #include <orc/orc.h>
 #endif
 
+#include <stdint.h>
+
 #ifndef _ORC_INTEGER_TYPEDEFS_
 #define _ORC_INTEGER_TYPEDEFS_
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
Index: schroedinger-1.0.10/testsuite/perf/schroorc-dist.c
===================================================================
--- schroedinger-1.0.10.orig/testsuite/perf/schroorc-dist.c
+++ schroedinger-1.0.10/testsuite/perf/schroorc-dist.c
@@ -8,6 +8,8 @@
 #include <orc/orc.h>
 #endif
 
+#include <stdint.h>
+
 #ifndef _ORC_INTEGER_TYPEDEFS_
 #define _ORC_INTEGER_TYPEDEFS_
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
