Index: attr-2.4.44/attr/Makefile
===================================================================
--- attr-2.4.44.orig/attr/Makefile
+++ attr-2.4.44/attr/Makefile
@@ -29,6 +29,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
Index: attr-2.4.44/getfattr/Makefile
===================================================================
--- attr-2.4.44.orig/getfattr/Makefile
+++ attr-2.4.44/getfattr/Makefile
@@ -30,6 +30,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
Index: attr-2.4.44/include/Makefile
===================================================================
--- attr-2.4.44.orig/include/Makefile
+++ attr-2.4.44/include/Makefile
@@ -33,7 +33,7 @@ $(INCDIR):
 include $(BUILDRULES)
 
 install-dev: default
-	$(INSTALL) -m 755 -d $(PKG_INC_DIR)
-	$(INSTALL) -m 644 $(INST_HFILES) $(PKG_INC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_INC_DIR)
+	$(INSTALL) -m 644 $(INST_HFILES) $(DESTDIR)$(PKG_INC_DIR)
 
 install install-lib:
Index: attr-2.4.44/man/man1/Makefile
===================================================================
--- attr-2.4.44.orig/man/man1/Makefile
+++ attr-2.4.44/man/man1/Makefile
@@ -22,7 +22,7 @@ include $(TOPDIR)/include/builddefs
 MAN_SECTION	= 1
 
 MAN_PAGES	= $(shell echo *.$(MAN_SECTION))
-MAN_DEST	= $(PKG_MAN_DIR)/man$(MAN_SECTION)
+MAN_DEST	= $(DESTDIR)$(PKG_MAN_DIR)/man$(MAN_SECTION)
 LSRCFILES	= $(MAN_PAGES)
 
 default : $(MAN_PAGES)
Index: attr-2.4.44/man/man2/Makefile
===================================================================
--- attr-2.4.44.orig/man/man2/Makefile
+++ attr-2.4.44/man/man2/Makefile
@@ -22,7 +22,7 @@ include $(TOPDIR)/include/builddefs
 MAN_SECTION	= 2
 
 MAN_PAGES	= $(shell echo *.$(MAN_SECTION))
-MAN_DEST	= $(PKG_MAN_DIR)/man$(MAN_SECTION)
+MAN_DEST	= $(DESTDIR)$(PKG_MAN_DIR)/man$(MAN_SECTION)
 LSRCFILES	= $(MAN_PAGES)
 
 default install : $(MAN_PAGES)
Index: attr-2.4.44/man/man3/Makefile
===================================================================
--- attr-2.4.44.orig/man/man3/Makefile
+++ attr-2.4.44/man/man3/Makefile
@@ -22,7 +22,7 @@ include $(TOPDIR)/include/builddefs
 MAN_SECTION	= 3
 
 MAN_PAGES	= $(shell echo *.$(MAN_SECTION))
-MAN_DEST	= $(PKG_MAN_DIR)/man$(MAN_SECTION)
+MAN_DEST	= $(DESTDIR)$(PKG_MAN_DIR)/man$(MAN_SECTION)
 LSRCFILES	= $(MAN_PAGES)
 
 default install : $(MAN_PAGES)
Index: attr-2.4.44/man/man5/Makefile
===================================================================
--- attr-2.4.44.orig/man/man5/Makefile
+++ attr-2.4.44/man/man5/Makefile
@@ -22,7 +22,7 @@ include $(TOPDIR)/include/builddefs
 MAN_SECTION	= 5
 
 MAN_PAGES	= $(shell echo *.$(MAN_SECTION))
-MAN_DEST	= $(PKG_MAN_DIR)/man$(MAN_SECTION)
+MAN_DEST	= $(DESTDIR)$(PKG_MAN_DIR)/man$(MAN_SECTION)
 LSRCFILES	= $(MAN_PAGES)
 
 default : $(MAN_PAGES)
Index: attr-2.4.44/setfattr/Makefile
===================================================================
--- attr-2.4.44.orig/setfattr/Makefile
+++ attr-2.4.44/setfattr/Makefile
@@ -30,6 +30,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
Index: attr-2.4.44/Makefile
===================================================================
--- attr-2.4.44.orig/Makefile
+++ attr-2.4.44/Makefile
@@ -89,13 +89,15 @@ include/config.h: include/builddefs
 		$(MAKE) $(AM_MAKEFLAGS) include/builddefs; \
 	fi
 
-install: default $(addsuffix -install,$(SUBDIRS))
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 README $(PKG_DOC_DIR)
+install: install-doc install-dev install-lib $(addsuffix -install,$(SUBDIRS))
+
+install-doc: default
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 README $(DESTDIR)$(PKG_DOC_DIR)
 
 install-dev: default $(addsuffix -install-dev,$(SUBDIRS))
 
-install-lib: install $(addsuffix -install-lib,$(SUBDIRS))
+install-lib: default $(addsuffix -install-lib,$(SUBDIRS))
 
 %-install:
 	$(MAKE) -C $* install
Index: attr-2.4.44/doc/Makefile
===================================================================
--- attr-2.4.44.orig/doc/Makefile
+++ attr-2.4.44/doc/Makefile
@@ -32,12 +32,12 @@ CHANGES.gz:
 
 install: default
 	$(SUBDIRS_MAKERULE)
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 PORTING CHANGES.gz $(PKG_DOC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 PORTING CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)
 ifeq ($(PKG_DISTRIBUTION), debian)
-	$(INSTALL) -S CHANGES.gz $(PKG_DOC_DIR)/changelog.gz
+	$(INSTALL) -S CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)/changelog.gz
 else
-	$(INSTALL) -m 644 COPYING $(PKG_DOC_DIR)
+	$(INSTALL) -m 644 COPYING $(DESTDIR)$(PKG_DOC_DIR)
 endif
 
 install-dev install-lib: $(SUBDIRS)
Index: attr-2.4.44/include/buildmacros
===================================================================
--- attr-2.4.44.orig/include/buildmacros
+++ attr-2.4.44/include/buildmacros
@@ -68,9 +68,9 @@ endif
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(PKG_LIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR)
 endif
 
 # Libtool thinks the static and shared libs should be in the same dir, so
@@ -81,25 +81,17 @@ endif
 # /usr/lib.
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB_DEV = \
-	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 $(LIBNAME).lai $(PKG_DEVLIB_DIR)/$(LIBNAME).la ; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_base $(LIBNAME).lai $(PKG_LIB_DIR); \
-	if test "x$(PKG_DEVLIB_DIR)" != "x$(PKG_LIB_DIR)" ; then \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).a $(PKG_LIB_DIR)/$(LIBNAME).a; \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).la $(PKG_LIB_DIR)/$(LIBNAME).la; \
-	../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).so $(PKG_DEVLIB_DIR)/$(LIBNAME).so; \
-	fi
+	cd $(TOPDIR)/$(LIBNAME)/.libs && \
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR) && \
+	../$(INSTALL) -T so_base $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR)
 else
 INSTALL_LTLIB_DEV = $(INSTALL_LTLIB_STATIC)
 endif
 
 INSTALL_LTLIB_STATIC = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DEVLIB_DIR); \
+	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(DESTDIR)$(PKG_DEVLIB_DIR)
 
 INSTALL_MAN = \
 	@for d in $(MAN_PAGES); do \
@@ -131,7 +123,7 @@ ifeq ($(ENABLE_GETTEXT),yes)
 INSTALL_LINGUAS = \
 	@for l in $(LINGUAS) ""; do \
 		if test -f "$$l.mo" ; then \
-			ldir=$(PKG_LOCALE_DIR)/$$l/LC_MESSAGES; \
+			ldir=$(DESTDIR)$(PKG_LOCALE_DIR)/$$l/LC_MESSAGES; \
 			$(INSTALL) -m 755 -d $$ldir; \
 			$(INSTALL) -m 644 $$l.mo $$ldir/$(PKG_NAME).mo; \
 		fi; \
