Index: lib/bb/fetch/__init__.py
===================================================================
--- lib/bb/fetch/__init__.py.orig
+++ lib/bb/fetch/__init__.py
@@ -443,6 +443,10 @@ def runfetchcmd(cmd, d, quiet = False):
                   'https_proxy', 'no_proxy', 'ALL_PROXY', 'all_proxy',
                   'KRB5CCNAME', 'SSH_AUTH_SOCK', 'SSH_AGENT_PID', 'HOME']
 
+    exportvars.extend(['https_proxy', 'no_proxy'])
+    exportvars.extend(['ELITO_CRT', 'ELITO_KEY', 'ELITO_PROG_SSL_CLIENT',
+                       'ELITO_PROG_PROXY', 'ELITO_USE_LIBPROXY'])
+
     for var in exportvars:
         val = data.getVar(var, d, True)
         if val:
