Index: lib/bb/data_smart.py
===================================================================
--- lib/bb/data_smart.py.orig
+++ lib/bb/data_smart.py
@@ -117,6 +117,8 @@ class DataSmart(MutableMapping):
                 s = __expand_python_regexp__.sub(varparse.python_sub, s)
                 if s == olds:
                     break
+            except bb.parse.SkipPackage:
+                raise
             except Exception:
                 logger.exception("Error evaluating '%s'", s)
                 raise
Index: lib/bb/parse/parse_py/BBHandler.py
===================================================================
--- lib/bb/parse/parse_py/BBHandler.py.orig
+++ lib/bb/parse/parse_py/BBHandler.py
@@ -150,7 +150,11 @@ def handle(fn, d, include):
     if ext != ".bbclass":
         data.setVar('FILE', fn, d)
 
-    statements.eval(d)
+    try:
+        statements.eval(d)
+    except bb.parse.SkipPackage:
+        bb.data.setVar("__SKIPPED", True, d)
+        return { "" : d }
 
     if ext == ".bbclass":
         classes.remove(__classname__)
