Index: setup.py
===================================================================
--- setup.py.orig
+++ setup.py
@@ -43,7 +43,7 @@ class Build(build):
         origpath = os.path.abspath(os.curdir)
         os.chdir(os.path.join(origpath, 'doc', 'manual'))
         make = os.environ.get('MAKE') or 'make'
-        ret = os.system('%s %s' % (make, doctype))
+        ret = 0
         if ret != 0:
             print("ERROR: Unable to generate html documentation.")
             sys.exit(ret)
