Index: udev-165/Makefile.am
===================================================================
--- udev-165.orig/Makefile.am
+++ udev-165/Makefile.am
@@ -304,10 +304,10 @@ libexec_PROGRAMS += extras/usb_id/usb_id
 # ------------------------------------------------------------------------------
 # v4l_id - video4linux capabilities
 # ------------------------------------------------------------------------------
-extras_v4l_id_v4l_id_SOURCES = extras/v4l_id/v4l_id.c
-extras_v4l_id_v4l_id_LDADD = libudev/libudev-private.la
-libexec_PROGRAMS += extras/v4l_id/v4l_id
-dist_udevrules_DATA += extras/v4l_id/60-persistent-v4l.rules
+#extras_v4l_id_v4l_id_SOURCES = extras/v4l_id/v4l_id.c
+#extras_v4l_id_v4l_id_LDADD = libudev/libudev-private.la
+#libexec_PROGRAMS += extras/v4l_id/v4l_id
+#dist_udevrules_DATA += extras/v4l_id/60-persistent-v4l.rules
 
 if ENABLE_EXTRAS
 # ------------------------------------------------------------------------------
