Index: udev-137/udev/udevd.c
===================================================================
--- udev-137.orig/udev/udevd.c
+++ udev-137/udev/udevd.c
@@ -644,10 +644,11 @@ int main(int argc, char *argv[])
 	int fd;
 	struct sigaction act;
 	const char *value;
-	int daemonize = 0;
+	enum { DAEMON_FG, DAEMON_BG, DAEMON_STOP } daemonize = DAEMON_FG;
 	int resolve_names = 1;
 	static const struct option options[] = {
 		{ "daemon", no_argument, NULL, 'd' },
+		{ "daemon-stop", no_argument, NULL, 'S' },
 		{ "debug-trace", no_argument, NULL, 't' },
 		{ "debug", no_argument, NULL, 'D' },
 		{ "help", no_argument, NULL, 'h' },
@@ -669,13 +670,16 @@ int main(int argc, char *argv[])
 	while (1) {
 		int option;
 
-		option = getopt_long(argc, argv, "dDthV", options, NULL);
+		option = getopt_long(argc, argv, "dDthVS", options, NULL);
 		if (option == -1)
 			break;
 
 		switch (option) {
 		case 'd':
-			daemonize = 1;
+			daemonize = DAEMON_BG;
+			break;
+		case 'S':
+			daemonize = DAEMON_STOP;
 			break;
 		case 't':
 			debug_trace = 1;
@@ -761,7 +765,9 @@ int main(int argc, char *argv[])
 	cleanup_queue_dir(udev);
 	export_initial_seqnum(udev);
 
-	if (daemonize) {
+	if (daemonize == DAEMON_STOP)
+		raise(SIGSTOP);
+	else if (daemonize == DAEMON_BG) {
 		pid_t pid;
 
 		pid = fork();
