Index: udev-146/Makefile.am
===================================================================
--- udev-146.orig/Makefile.am
+++ udev-146/Makefile.am
@@ -503,8 +503,8 @@ keymaps-distcheck-hook: extras/keymap/ke
 # Custom rules
 # ------------------------------------------------------------------------------
 
-extras/keymap/keys.txt: /usr/include/linux/input.h
-	$(AM_V_GEN)$(AWK) '/^#define.*KEY_/ { if ($$2 != "KEY_MAX" && $$2 != "KEY_CNT") { print $$2 } }' < $< > $@
+extras/keymap/keys.txt:
+	$(AM_V_GEN)echo '#include <linux/input.h>' | $(CC) $(CFLAGS) -E -dD - | $(AWK) '/^#define[ \t]+KEY_/ { if ($$2 != "KEY_MAX" && $$2 != "KEY_CNT") { print $$2 } }' > $@
 
 extras/keymap/keys-from-name.gperf: extras/keymap/keys.txt
 	$(AM_V_GEN)$(AWK) 'BEGIN{ print "struct key { const char* name; unsigned short id; };"; print "%null-strings"; print "%%";} { print $$1 ", " $$1 }' < $< > $@
