Index: udev-145/udev/udevd.c
===================================================================
--- udev-145.orig/udev/udevd.c
+++ udev-145/udev/udevd.c
@@ -776,10 +776,11 @@ int main(int argc, char *argv[])
 	int fd;
 	sigset_t mask;
 	const char *value;
-	int daemonize = false;
+	enum { DAEMON_FG, DAEMON_BG, DAEMON_STOP } daemonize = DAEMON_FG;
 	int resolve_names = 1;
 	static const struct option options[] = {
 		{ "daemon", no_argument, NULL, 'd' },
+		{ "daemon-stop", no_argument, NULL, 'S' },
 		{ "debug-trace", no_argument, NULL, 't' },
 		{ "debug", no_argument, NULL, 'D' },
 		{ "help", no_argument, NULL, 'h' },
@@ -801,13 +802,16 @@ int main(int argc, char *argv[])
 	while (1) {
 		int option;
 
-		option = getopt_long(argc, argv, "dDthV", options, NULL);
+		option = getopt_long(argc, argv, "dDthVS", options, NULL);
 		if (option == -1)
 			break;
 
 		switch (option) {
 		case 'd':
-			daemonize = true;
+			daemonize = DAEMON_BG;
+			break;
+		case 'S':
+			daemonize = DAEMON_STOP;
 			break;
 		case 't':
 			debug_trace = true;
@@ -944,7 +948,9 @@ int main(int argc, char *argv[])
 		goto exit;
 	}
 
-	if (daemonize) {
+	if (daemonize == DAEMON_STOP)
+		raise(SIGSTOP);
+	else if (daemonize == DAEMON_BG) {
 		pid_t pid;
 
 		pid = fork();
