Index: upstart-0.6.3/init/job_process.c
===================================================================
--- upstart-0.6.3.orig/init/job_process.c
+++ upstart-0.6.3/init/job_process.c
@@ -57,6 +57,32 @@
 #include "errors.h"
 
 
+#if defined __GLIBC_PREREQ && ! __GLIBC_PREREQ(2,9)
+#  warning Your <sys/ptrace.h> is too old; defining __ptrace_* enums here...
+
+/* Options set using PTRACE_SETOPTIONS.  */
+enum __ptrace_setoptions {
+  PTRACE_O_TRACESYSGOOD = 0x00000001,
+  PTRACE_O_TRACEFORK    = 0x00000002,
+  PTRACE_O_TRACEVFORK   = 0x00000004,
+  PTRACE_O_TRACECLONE   = 0x00000008,
+  PTRACE_O_TRACEEXEC    = 0x00000010,
+  PTRACE_O_TRACEVFORKDONE = 0x00000020,
+  PTRACE_O_TRACEEXIT    = 0x00000040,
+  PTRACE_O_MASK         = 0x0000007f
+};
+
+/* Wait extended result codes for the above trace options.  */
+enum __ptrace_eventcodes {
+  PTRACE_EVENT_FORK     = 1,
+  PTRACE_EVENT_VFORK    = 2,
+  PTRACE_EVENT_CLONE    = 3,
+  PTRACE_EVENT_EXEC     = 4,
+  PTRACE_EVENT_VFORK_DONE = 5,
+  PTRACE_EVENT_EXIT     = 6
+};
+#endif
+
 /**
  * SHELL_CHARS:
  *
