From 89cc9633a581381f4547c7b94ef63b56937ab9cd Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
Date: Mon, 17 Jun 2013 20:12:09 +0200
Subject: [PATCH 09/10] install bitbake-worker

Signed-off-by: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index aff42a8..17d1453 100755
--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(name='bitbake',
                   "bb.server", "bb.parse.parse_py", "bb.parse",
                   "bb.fetch2", "bb.ui.crumbs", "bb.ui", "bb.pysh", "bb", "prserv", "bb.tests"],
       py_modules = ["codegen", "progressbar"],
-      scripts = ["bin/bitbake", "bin/bitbake-layers", "bin/bitbake-diffsigs", "bin/bitbake-prserv", "bin/bitbake-selftest", "bin/image-writer"],
+      scripts = ["bin/bitbake", "bin/bitbake-layers", "bin/bitbake-diffsigs", "bin/bitbake-prserv", "bin/bitbake-selftest", "bin/image-writer", "bin/bitbake-worker"],
       data_files = [("share/bitbake", glob("conf/*") + glob("classes/*")),
                   ("share/doc/bitbake-%s/bitbake-user-manual" % __version__, glob("doc/bitbake-user-manual/html/*"))],
       cmdclass = {
-- 
1.9.0

