From 4bbc61e23f523d81f0120d5d189480e190b9b589 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:42 +0200
Subject: [PATCH 2/8] export some more proxy related environment variables

---
 lib/bb/fetch2/__init__.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/bb/fetch2/__init__.py b/lib/bb/fetch2/__init__.py
index 136fc29..4d857d8 100644
--- a/lib/bb/fetch2/__init__.py
+++ b/lib/bb/fetch2/__init__.py
@@ -820,6 +820,9 @@ def runfetchcmd(cmd, d, quiet=False, cleanup=None, log=None, workdir=None):
                   'DBUS_SESSION_BUS_ADDRESS',
                   'P4CONFIG']
 
+    exportvars.extend(['ELITO_CRT', 'ELITO_KEY', 'ELITO_PROG_SSL_CLIENT',
+                       'ELITO_PROG_PROXY', 'ELITO_USE_LIBPROXY'])
+
     if not cleanup:
         cleanup = []
 
-- 
2.9.4

