From 9c3a6342702289d3bfb88d9330de3dc7baab24e4 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:44 +0200
Subject: [PATCH 3/8] import 're' module by default

---
 lib/bb/utils.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/bb/utils.py b/lib/bb/utils.py
index 077fddc..50d50f9 100644
--- a/lib/bb/utils.py
+++ b/lib/bb/utils.py
@@ -51,6 +51,7 @@ def clean_context():
         "os": os,
         "bb": bb,
         "time": time,
+        "re": re,
     }
 
 def get_context():
-- 
2.9.4

