From 30e8276466b36695114c033e03ba0c9d08a5207e Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:47 +0200
Subject: [PATCH 4/4] do not execute the 'make doc' target

Signed-off-by: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 8c26a9b..5971bc9 100755
--- a/setup.py
+++ b/setup.py
@@ -40,6 +40,7 @@ class Clean(clean):
 class Build(build):
     def run(self):
         build.run(self)
+        return 0
         origpath = os.path.abspath(os.curdir)
         os.chdir(os.path.join(origpath, 'doc', 'manual'))
         make = os.environ.get('MAKE') or 'make'
-- 
1.8.5.3

