From 8f39a72163320a77dd79d517bc0061b5d831f72a Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:44 +0200
Subject: [PATCH 03/10] import 're' module by default

---
 lib/bb/utils.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/bb/utils.py b/lib/bb/utils.py
index 7686602..8c9fd75 100644
--- a/lib/bb/utils.py
+++ b/lib/bb/utils.py
@@ -41,6 +41,7 @@ def clean_context():
         "os": os,
         "bb": bb,
         "time": time,
+        "re": re,
     }
 
 def get_context():
-- 
1.8.1.4

