From 478f803379e2f2502b73c65c75c2d31aef97fa91 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:42 +0200
Subject: [PATCH 05/16] export some more proxy related environment variables

---
 lib/bb/fetch2/__init__.py |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/lib/bb/fetch2/__init__.py b/lib/bb/fetch2/__init__.py
index 168efb5..1b8e044 100644
--- a/lib/bb/fetch2/__init__.py
+++ b/lib/bb/fetch2/__init__.py
@@ -407,6 +407,9 @@ def runfetchcmd(cmd, d, quiet = False, cleanup = []):
                   'SSH_AUTH_SOCK', 'SSH_AGENT_PID', 'HOME',
                   'GIT_PROXY_IGNORE', 'SOCKS5_USER', 'SOCKS5_PASSWD']
 
+    exportvars.extend(['ELITO_CRT', 'ELITO_KEY', 'ELITO_PROG_SSL_CLIENT',
+                       'ELITO_PROG_PROXY', 'ELITO_USE_LIBPROXY'])
+
     new_env = os.environ.copy()
     for var in exportvars:
         val = d.getVar(var, True)
-- 
1.7.7.6

