From 3c68908620d2f5931e10b7683a19434e3418b10c Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 8 Oct 2011 12:57:24 +0200
Subject: [PATCH 06/11] added backward compatibility for 'git remote add'

---
 lib/bb/fetch2/git.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/bb/fetch2/git.py b/lib/bb/fetch2/git.py
index b4b9368..87e5ba2 100644
--- a/lib/bb/fetch2/git.py
+++ b/lib/bb/fetch2/git.py
@@ -118,6 +118,7 @@ class Git(FetchMethod):
 
         ud.basecmd = data.getVar("FETCHCMD_git", d, True) or "git"
 
+        ud.mirroropt = data.getVar("FETCHCMD_OPT_GIT_MIRROR", d, True) or "--mirror=fetch"
         ud.write_tarballs = ((data.getVar("BB_GENERATE_MIRROR_TARBALLS", d, True) or "0") != "0") or ud.rebaseable
 
         ud.setup_revisons(d)
@@ -207,7 +208,7 @@ class Git(FetchMethod):
             except bb.fetch2.FetchError:
                 logger.debug(1, "No Origin")
 
-            runfetchcmd("%s remote add --mirror=fetch origin %s" % (ud.basecmd, repourl), d)
+            runfetchcmd("%s remote add %s origin %s" % (ud.basecmd, ud.mirroropt, repourl), d)
             fetch_cmd = "%s fetch -f --prune %s refs/*:refs/*" % (ud.basecmd, repourl)
             if ud.proto.lower() != 'file':
                 bb.fetch2.check_network_access(d, fetch_cmd, ud.url)
-- 
1.8.3.1

