From b84e67b3f96212e15c17f12eb015785eaa4ee7b5 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
Date: Tue, 21 May 2013 19:17:21 +0200
Subject: [PATCH 6/8] fetcher: provide a way to rewrite the SRC_URI very early

Signed-off-by: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
---
 lib/bb/fetch2/__init__.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/bb/fetch2/__init__.py b/lib/bb/fetch2/__init__.py
index 4d857d8..4a953b2 100644
--- a/lib/bb/fetch2/__init__.py
+++ b/lib/bb/fetch2/__init__.py
@@ -1197,6 +1197,7 @@ class FetchData(object):
         self.basename = None
         self.basepath = None
         (self.type, self.host, self.path, self.user, self.pswd, self.parm) = decodeurl(d.expand(url))
+        self.__rewrite_url(d)
         self.date = self.getSRCDate(d)
         self.url = url
         if not self.user and "user" in self.parm:
@@ -1271,6 +1272,12 @@ class FetchData(object):
         self.donestamp = basepath + '.done'
         self.lockfile = basepath + '.lock'

+    def __rewrite_url(self, d):
+        fn = d.getVarFlag('SRC_URI', '_rewrite_fn', True, False)
+        if fn:
+            codeobj = compile("%s(data, d)" % fn, 'SRC_URI', 'eval')
+            bb.utils.better_eval(codeobj, { 'd' : d, 'data' : self })
+
     def setup_revisions(self, d):
         self.revisions = {}
         for name in self.names:
--
2.9.4
