From 7c90d9e1fbee596cf48c1918fd55e763ccc45abb Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:48 +0200
Subject: [PATCH 06/12] install ply + progressbar modules

---
 setup.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f760e88..c4fb10e 100755
--- a/setup.py
+++ b/setup.py
@@ -53,9 +53,10 @@ setup(name='bitbake',
       version = __version__,
       requires = ["ply", "progressbar"],
       package_dir = {"": "lib"},
-      packages = ["bb.server", "bb.parse.parse_py", "bb.parse", 
+      packages = ["ply",
+                  "bb.server", "bb.parse.parse_py", "bb.parse",
                   "bb.fetch2", "bb.ui.crumbs", "bb.ui", "bb.pysh", "bb", "prserv", "bb.tests"],
-      py_modules = ["codegen"],
+      py_modules = ["codegen", "progressbar"],
       scripts = ["bin/bitbake", "bin/bitbake-layers", "bin/bitbake-diffsigs", "bin/bitbake-prserv", "bin/bitbake-selftest", "bin/image-writer"],
       data_files = [("share/bitbake", glob("conf/*") + glob("classes/*")),
                   ("share/doc/bitbake-%s/manual" % __version__, glob("doc/manual/html/*"))],
-- 
1.7.11.7

