From 324de226368406bdacc90bb939e88c53dab92411 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:44 +0200
Subject: [PATCH 08/19] import 're' module by default

---
 lib/bb/utils.py |    1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/bb/utils.py b/lib/bb/utils.py
index fc389a3..1fd0dcb 100644
--- a/lib/bb/utils.py
+++ b/lib/bb/utils.py
@@ -36,6 +36,7 @@ _context = {
     "os": os,
     "bb": bb,
     "time": time,
+    "re": re,
 }
 
 def explode_version(s):
-- 
1.7.10.2

