From 3da79e3aa448be41e8c46bcca6f5970b8a71a2c1 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:47 +0200
Subject: [PATCH 09/16] detect and ignore duplicate inherits

---
 lib/bb/cooker.py |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/lib/bb/cooker.py b/lib/bb/cooker.py
index 7297cf4..123d0c1 100644
--- a/lib/bb/cooker.py
+++ b/lib/bb/cooker.py
@@ -858,7 +858,13 @@ class BBCooker:
 
         # Handle any INHERITs and inherit the base class
         bbclasses  = ["base"] + (data.getVar('INHERIT', True) or "").split()
+        seen_inherits={}
         for bbclass in bbclasses:
+            if seen_inherits.has_key(bbclass):
+                parselog.warn("Duplicate INHERIT entry '%s'" % bbclass)
+                continue
+
+            seen_inherits[bbclass] = 1
             data = _inherit(bbclass, data)
 
         # Nomally we only register event handlers at the end of parsing .bb files
-- 
1.7.7.6

