From 09e4f816c66ec0e2ecf59dd52fdc63c12e0d2ae8 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:47 +0200
Subject: [PATCH 09/12] detect and ignore duplicate inherits

---
 lib/bb/cooker.py |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/lib/bb/cooker.py b/lib/bb/cooker.py
index 47389e8..5e34e4f 100644
--- a/lib/bb/cooker.py
+++ b/lib/bb/cooker.py
@@ -805,7 +805,13 @@ class BBCooker:
 
         # Handle any INHERITs and inherit the base class
         bbclasses  = ["base"] + (data.getVar('INHERIT', True) or "").split()
+        seen_inherits={}
         for bbclass in bbclasses:
+            if seen_inherits.has_key(bbclass):
+                parselog.warn("Duplicate INHERIT entry '%s'" % bbclass)
+                continue
+
+            seen_inherits[bbclass] = 1
             data = _inherit(bbclass, data)
 
         # Nomally we only register event handlers at the end of parsing .bb files
-- 
1.7.6

