From 006621202ee740bb9b10e92daaf28911a9604abb Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:46 +0200
Subject: [PATCH 10/20] added BBMASK_BASENAMES variables

---
 lib/bb/cooker.py |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/lib/bb/cooker.py b/lib/bb/cooker.py
index 72163e9..bccf06a 100644
--- a/lib/bb/cooker.py
+++ b/lib/bb/cooker.py
@@ -1343,6 +1343,8 @@ class BBCooker:
                         newfiles.append(g)
 
         bbmask = self.configuration.data.getVar('BBMASK', True)
+        bbmask_basenames = (self.configuration.data.getVar('BBMASK_BASENAMES', True)
+                            or "").split()
 
         if bbmask:
             try:
@@ -1350,12 +1352,23 @@ class BBCooker:
             except sre_constants.error:
                 collectlog.critical("BBMASK is not a valid regular expression, ignoring.")
                 return list(newfiles), 0
+        elif not bbmask_basenames:
+            return list(newfiles), 0
+
+        bbmask_dict = {}
+        for f in bbmask_basenames:
+            if not f.endswith('.bb'):
+                f += '.bb'
+
+            bbmask_dict[f] = 1
 
         bbfiles = []
         bbappend = []
         maskedfiles = []
         for f in newfiles:
-            if bbmask and bbmask_compiled.search(f):
+            f_base = os.path.basename(f)
+            if bbmask_dict.has_key(f_base) or (bbmask and
+                                               bbmask_compiled.search(f)):
                 collectlog.debug(2, "skipping masked file %s", f)
                 if collectlog.level >= 1:
                     maskedfiles.append(f)
-- 
1.7.10.4

