From ddbac2212e57259fae0da3db3db79c5591ea2a1b Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:47 +0200
Subject: [PATCH 11/16] do not execute the 'make doc' target

---
 setup.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/setup.py b/setup.py
index bc2815d..e103779 100755
--- a/setup.py
+++ b/setup.py
@@ -43,7 +43,7 @@ class Build(build):
         origpath = os.path.abspath(os.curdir)
         os.chdir(os.path.join(origpath, 'doc', 'manual'))
         make = os.environ.get('MAKE') or 'make'
-        ret = os.system('%s %s' % (make, doctype))
+        ret = 0
         if ret != 0:
             print("ERROR: Unable to generate html documentation.")
             sys.exit(ret)
-- 
1.7.7.6

