Index: bitbake/lib/bb/data_smart.py
===================================================================
--- lib/bb/data_smart.py
+++ lib/bb/data_smart.py
@@ -40,6 +40,7 @@ __setvar_regexp__ = re.compile('(?P<base
 __expand_var_regexp__ = re.compile(r"\${[^{}]+}")
 __expand_python_regexp__ = re.compile(r"\${@.+?}")
 _expand_globals = {
+    "re": re,
     "os": os,
     "bb": bb,
     "time": time,
