From 1b1eb9a7c8db5929c2d910442b1d6ef1374f12fb Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 8 Oct 2011 12:57:24 +0200
Subject: [PATCH 15/16] added backward compatibility for 'git remote add'

---
 lib/bb/fetch2/git.py |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/lib/bb/fetch2/git.py b/lib/bb/fetch2/git.py
index c1b6904..86140be 100644
--- a/lib/bb/fetch2/git.py
+++ b/lib/bb/fetch2/git.py
@@ -202,7 +202,11 @@ class Git(FetchMethod):
             except bb.fetch2.FetchError:
                 logger.debug(1, "No Origin")
 
-            ud.runfetch(['remote', 'add', '--mirror=fetch', 'origin', repourl], d)
+            try:
+                ud.runfetch(['remote', 'add', '--mirror=fetch', 'origin', repourl], d)
+            except bb.fetch2.FetchError:
+                ud.runfetch(['remote', 'add', '--mirror', 'origin', repourl], d)
+
             fetch_cmd = ['fetch', '-f', '--prune', repourl, 'refs/*:refs/*']
             bb.fetch2.check_network_access(d, fetch_cmd, base_cmd = ud.basecmd, url = ud.url)
             ud.runfetch(fetch_cmd, d)
-- 
1.7.7.6

