From c8c3d68eb7bd5e2a1a5a19f9e445307f99eed0ba Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
Date: Mon, 21 May 2012 17:15:44 +0200
Subject: [PATCH 15/17] siggen: dump sigdata content

---
 lib/bb/siggen.py | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/lib/bb/siggen.py b/lib/bb/siggen.py
index 8fe59b9..c9a5ad7 100644
--- a/lib/bb/siggen.py
+++ b/lib/bb/siggen.py
@@ -63,6 +63,7 @@ class SignatureGeneratorBasic(SignatureGenerator):
     def __init__(self, data):
         self.basehash = {}
         self.taskhash = {}
+        self.taskhash_cont = {}
         self.taskdeps = {}
         self.runtaskdeps = {}
         self.file_checksum_values = {}
@@ -170,6 +171,7 @@ class SignatureGeneratorBasic(SignatureGenerator):
     def get_taskhash(self, fn, task, deps, dataCache):
         k = fn + "." + task
         data = dataCache.basetaskhash[k]
+        cont = [('__base', str(data))]
         self.runtaskdeps[k] = []
         self.file_checksum_values[k] = {}
         recipename = dataCache.pkg_fn[fn]
@@ -180,20 +182,24 @@ class SignatureGeneratorBasic(SignatureGenerator):
             if dep not in self.taskhash:
                 bb.fatal("%s is not in taskhash, caller isn't calling in dependency order?", dep)
             data = data + self.taskhash[dep]
+            cont.append([str(dep), self.taskhash[dep]])
             self.runtaskdeps[k].append(dep)
 
         if task in dataCache.file_checksums[fn]:
             checksums = bb.fetch2.get_file_checksums(dataCache.file_checksums[fn][task], recipename)
             for (f,cs) in checksums:
-                self.file_checksum_values[k][f] = cs
-                data = data + cs
+               self.file_checksum_values[k][f] = cs
+               data = data + cs
+               cont.append([str(f), cs])
 
         taint = self.read_taint(fn, task, dataCache.stamp[fn])
         if taint:
             data = data + taint
+            cont.append(['TAINT', taint])
 
         h = hashlib.md5(data).hexdigest()
         self.taskhash[k] = h
+        self.taskhash_cont[k] = cont
         #d.setVar("BB_TASKHASH_task-%s" % task, taskhash[task])
         return h
 
@@ -252,6 +258,17 @@ class SignatureGeneratorBasic(SignatureGenerator):
                 pass
             raise err
 
+        fd, tmpfile = tempfile.mkstemp(dir=os.path.dirname(sigfile),
+                                       prefix="sigtask.txt.")
+        try:
+            with os.fdopen(fd, "wb") as stream:
+                stream.write(''.join(map(lambda x: '%s: %s\n' %
+                                         (x[0],x[1]), self.taskhash_cont[k])))
+            os.chmod(tmpfile, 0664)
+            os.rename(tmpfile, sigfile.replace('.sigdata.', '.sigtext.').replace('.sigbasedata.', '.sigbasetext.'))
+        finally:
+            bb.utils.remove(tmpfile)
+
     def dump_sigs(self, dataCache):
         for fn in self.taskdeps:
             for task in self.taskdeps[fn]:
-- 
1.7.11.4

