From 784971088ebb36085261eb55fce8f51b4483f58e Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@sigma-chemnitz.de>
Date: Mon, 21 May 2012 17:15:44 +0200
Subject: [PATCH 19/19] siggen: dump sigdata content

---
 lib/bb/siggen.py |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/bb/siggen.py b/lib/bb/siggen.py
index edd98fc..06adb59 100644
--- a/lib/bb/siggen.py
+++ b/lib/bb/siggen.py
@@ -62,6 +62,7 @@ class SignatureGeneratorBasic(SignatureGenerator):
     def __init__(self, data):
         self.basehash = {}
         self.taskhash = {}
+        self.taskhash_cont = {}
         self.taskdeps = {}
         self.runtaskdeps = {}
         self.file_checksum_values = {}
@@ -169,6 +170,7 @@ class SignatureGeneratorBasic(SignatureGenerator):
     def get_taskhash(self, fn, task, deps, dataCache):
         k = fn + "." + task
         data = dataCache.basetaskhash[k]
+        cont = [('__base', str(data))]
         self.runtaskdeps[k] = []
         self.file_checksum_values[k] = {}
         recipename = dataCache.pkg_fn[fn]
@@ -179,6 +181,7 @@ class SignatureGeneratorBasic(SignatureGenerator):
             if dep not in self.taskhash:
                 bb.fatal("%s is not in taskhash, caller isn't calling in dependency order?", dep)
             data = data + self.taskhash[dep]
+            cont.append([str(dep), self.taskhash[dep]])
             self.runtaskdeps[k].append(dep)
 
         if task in dataCache.file_checksums[fn]:
@@ -186,13 +189,16 @@ class SignatureGeneratorBasic(SignatureGenerator):
             for (f,cs) in checksums:
                self.file_checksum_values[k][f] = cs
                data = data + cs
+               cont.append([str(f), cs])
 
         taint = self.read_taint(fn, task, dataCache.stamp[fn])
         if taint:
             data = data + taint
+            cont.append(['TAINT', cs])
 
         h = hashlib.md5(data).hexdigest()
         self.taskhash[k] = h
+        self.taskhash_cont[k] = cont
         #d.setVar("BB_TASKHASH_task-%s" % task, taskhash[task])
         return h
 
@@ -239,6 +245,10 @@ class SignatureGeneratorBasic(SignatureGenerator):
         with open(sigfile, "wb") as f:
             p = pickle.Pickler(f, -1)
             p.dump(data)
+
+        open(sigfile + ".txt", "wb").write(
+            ''.join(map(lambda x: '%s: %s\n' % (x[0],x[1]), self.taskhash_cont[k])))
+
         os.chmod(sigfile, 0664)
 
     def dump_sigs(self, dataCache):
-- 
1.7.10.2

