Index: acl-2.2.48/include/builddefs.in
===================================================================
--- acl-2.2.48.orig/include/builddefs.in
+++ acl-2.2.48/include/builddefs.in
@@ -16,6 +16,7 @@ LIBACL = $(TOPDIR)/libacl/libacl.la
 LIBATTR = @libattr@
 LIBMISC = $(TOPDIR)/libmisc/libmisc.la
 
+top_builddir = @abs_top_builddir@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 top_builddir = @top_builddir@
Index: acl-2.2.48/Makefile
===================================================================
--- acl-2.2.48.orig/Makefile
+++ acl-2.2.48/Makefile
@@ -88,9 +88,13 @@ include/config.h: include/builddefs
 		$(MAKE) $(AM_MAKEFLAGS) include/builddefs; \
 	fi
 
-install: default $(addsuffix -install,$(SUBDIRS))
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 README $(PKG_DOC_DIR)
+install: install-dev install-lib install-doc install-common
+
+install-doc: default
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 README $(DESTDIR)$(PKG_DOC_DIR)
+
+install-common: default $(addsuffix -install,$(SUBDIRS))
 
 install-dev: default $(addsuffix -install-dev,$(SUBDIRS))
 
Index: acl-2.2.48/chacl/Makefile
===================================================================
--- acl-2.2.48.orig/chacl/Makefile
+++ acl-2.2.48/chacl/Makefile
@@ -30,6 +30,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
Index: acl-2.2.48/doc/Makefile
===================================================================
--- acl-2.2.48.orig/doc/Makefile
+++ acl-2.2.48/doc/Makefile
@@ -31,12 +31,12 @@ CHANGES.gz:
 	$(ZIP) --best -c < CHANGES > $@
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 PORTING CHANGES.gz $(PKG_DOC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 PORTING CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)
 ifeq ($(PKG_DISTRIBUTION), debian)
-	$(INSTALL) -S CHANGES.gz $(PKG_DOC_DIR)/changelog.gz
+	$(INSTALL) -S CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)/changelog.gz
 else
-	$(INSTALL) -m 644 COPYING COPYING.LGPL $(PKG_DOC_DIR)
+	$(INSTALL) -m 644 COPYING COPYING.LGPL $(DESTDIR)$(PKG_DOC_DIR)
 endif
 
 install-dev install-lib:
Index: acl-2.2.48/getfacl/Makefile
===================================================================
--- acl-2.2.48.orig/getfacl/Makefile
+++ acl-2.2.48/getfacl/Makefile
@@ -31,6 +31,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
Index: acl-2.2.48/include/Makefile
===================================================================
--- acl-2.2.48.orig/include/Makefile
+++ acl-2.2.48/include/Makefile
@@ -31,10 +31,10 @@ sys acl:
 include $(BUILDRULES)
 
 install-dev: default
-	$(INSTALL) -m 755 -d $(PKG_INC_DIR)
-	$(INSTALL) -m 755 -d $(PKG_INC_DIR)/sys
-	$(INSTALL) -m 644 acl.h $(PKG_INC_DIR)/sys/acl.h
-	$(INSTALL) -m 755 -d $(PKG_INC_DIR)/acl
-	$(INSTALL) -m 644 libacl.h $(PKG_INC_DIR)/acl/libacl.h
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_INC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_INC_DIR)/sys
+	$(INSTALL) -m 644 acl.h $(DESTDIR)$(PKG_INC_DIR)/sys/acl.h
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_INC_DIR)/acl
+	$(INSTALL) -m 644 libacl.h $(DESTDIR)$(PKG_INC_DIR)/acl/libacl.h
 
 install install-lib:
Index: acl-2.2.48/include/buildmacros
===================================================================
--- acl-2.2.48.orig/include/buildmacros
+++ acl-2.2.48/include/buildmacros
@@ -68,9 +68,9 @@ endif
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(PKG_LIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR)
 endif
 
 # Libtool thinks the static and shared libs should be in the same dir, so
@@ -81,25 +81,17 @@ endif
 # /usr/lib.
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB_DEV = \
-	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 $(LIBNAME).lai $(PKG_DEVLIB_DIR)/$(LIBNAME).la ; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_base $(LIBNAME).lai $(PKG_LIB_DIR); \
-	if test "x$(PKG_DEVLIB_DIR)" != "x$(PKG_LIB_DIR)" ; then \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).a $(PKG_LIB_DIR)/$(LIBNAME).a; \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).la $(PKG_LIB_DIR)/$(LIBNAME).la; \
-	../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).so $(PKG_DEVLIB_DIR)/$(LIBNAME).so; \
-	fi
+	cd $(TOPDIR)/$(LIBNAME)/.libs && \
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR) && \
+	../$(INSTALL) -T so_base $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR)
 else
 INSTALL_LTLIB_DEV = $(INSTALL_LTLIB_STATIC)
 endif
 
 INSTALL_LTLIB_STATIC = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DEVLIB_DIR); \
+	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(DESTDIR)$(PKG_DEVLIB_DIR)
 
 INSTALL_MAN = \
 	@for d in $(MAN_PAGES); do \
@@ -111,7 +103,7 @@ INSTALL_MAN = \
 				-e 's/\\\f[0-9]//g' -e 's/  / /g;q'`; \
 		do \
 			[ -z "$$m" -o "$$m" = "\\" ] && continue; \
-			t=$(MAN_DEST)/$$m.$(MAN_SECTION); \
+			t=$(DESTDIR)$(MAN_DEST)/$$m.$(MAN_SECTION); \
 			if $$first; then \
 				if $(HAVE_ZIPPED_MANPAGES); then \
 					$(ZIP) -9 -c $$d > $$d.gz; _sfx=.gz; \
@@ -131,7 +123,7 @@ ifeq ($(ENABLE_GETTEXT),yes)
 INSTALL_LINGUAS = \
 	@for l in $(LINGUAS) ""; do \
 		if test -f "$$l.mo" ; then \
-			ldir=$(PKG_LOCALE_DIR)/$$l/LC_MESSAGES; \
+			ldir=$(DESTDIR)$(PKG_LOCALE_DIR)/$$l/LC_MESSAGES; \
 			$(INSTALL) -m 755 -d $$ldir; \
 			$(INSTALL) -m 644 $$l.mo $$ldir/$(PKG_NAME).mo; \
 		fi; \
Index: acl-2.2.48/man/man1/Makefile
===================================================================
--- acl-2.2.48.orig/man/man1/Makefile
+++ acl-2.2.48/man/man1/Makefile
@@ -30,6 +30,6 @@ default : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-dev install-lib:
Index: acl-2.2.48/man/man3/Makefile
===================================================================
--- acl-2.2.48.orig/man/man3/Makefile
+++ acl-2.2.48/man/man3/Makefile
@@ -30,6 +30,6 @@ default : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install-dev : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install install-lib:
Index: acl-2.2.48/man/man5/Makefile
===================================================================
--- acl-2.2.48.orig/man/man5/Makefile
+++ acl-2.2.48/man/man5/Makefile
@@ -30,6 +30,6 @@ default : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-dev install-lib:
Index: acl-2.2.48/setfacl/Makefile
===================================================================
--- acl-2.2.48.orig/setfacl/Makefile
+++ acl-2.2.48/setfacl/Makefile
@@ -31,6 +31,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
