Index: git/tests/ts_calibrate.c
===================================================================
--- git.orig/tests/ts_calibrate.c
+++ git/tests/ts_calibrate.c
@@ -204,7 +204,7 @@ int main()
 	if( (tsdevice = getenv("TSLIB_TSDEVICE")) != NULL ) {
 		ts = ts_open(tsdevice,0);
 	} else {
-		if (!(ts = ts_open("/dev/input/event0", 0)))
+		if (!(ts = ts_open("/dev/input/event-touchscreen0", 0)))
 			ts = ts_open("/dev/touchscreen/ucb1x00", 0);
 	}
 
Index: git/tests/ts_harvest.c
===================================================================
--- git.orig/tests/ts_harvest.c
+++ git/tests/ts_harvest.c
@@ -76,7 +76,7 @@ int main()
 	if( (tsdevice = getenv("TSLIB_TSDEVICE")) != NULL ) {
 		ts = ts_open(tsdevice,0);
 	} else {
-		if (!(ts = ts_open("/dev/input/event0", 0)))
+		if (!(ts = ts_open("/dev/input/event-touchscreen0", 0)))
 			ts = ts_open("/dev/touchscreen/ucb1x00", 0);
 	}
 
Index: git/tests/ts_print.c
===================================================================
--- git.orig/tests/ts_print.c
+++ git/tests/ts_print.c
@@ -28,7 +28,7 @@ int main()
 	if( (tsdevice = getenv("TSLIB_TSDEVICE")) != NULL ) {
 		ts = ts_open(tsdevice,0);
 	} else {
-		if (!(ts = ts_open("/dev/input/event0", 0)))
+		if (!(ts = ts_open("/dev/input/event-touchscreen0", 0)))
 			ts = ts_open("/dev/touchscreen/ucb1x00", 0);
 	}
 
Index: git/tests/ts_print_raw.c
===================================================================
--- git.orig/tests/ts_print_raw.c
+++ git/tests/ts_print_raw.c
@@ -28,7 +28,7 @@ int main()
 	if( (tsdevice = getenv("TSLIB_TSDEVICE")) != NULL ) {
 		ts = ts_open(tsdevice,0);
 	} else {
-		if (!(ts = ts_open("/dev/input/event0", 0)))
+		if (!(ts = ts_open("/dev/input/event-touchscreen0", 0)))
 			ts = ts_open("/dev/touchscreen/ucb1x00", 0);
 	}
 
Index: git/tests/ts_test.c
===================================================================
--- git.orig/tests/ts_test.c
+++ git/tests/ts_test.c
@@ -122,7 +122,7 @@ int main()
 	if( (tsdevice = getenv("TSLIB_TSDEVICE")) != NULL ) {
 		ts = ts_open(tsdevice,0);
 	} else {
-		if (!(ts = ts_open("/dev/input/event0", 0)))
+		if (!(ts = ts_open("/dev/input/event-touchscreen0", 0)))
 			ts = ts_open("/dev/touchscreen/ucb1x00", 0);
 	}
 
