Index: sysvinit-2.86/src/bootlogd.c
===================================================================
--- sysvinit-2.86.orig/src/bootlogd.c
+++ sysvinit-2.86/src/bootlogd.c
@@ -230,7 +230,11 @@ int consolename(char *res, int rlen)
 #ifdef TIOCGDEV
 	if (ioctl(0, TIOCGDEV, &kdev) == 0)
 		return findtty(res, rlen, (dev_t)kdev);
-	if (errno != ENOIOCTLCMD) return -1;
+	if (
+#  ifdef ENOIOCTLCMD
+		errno != ENOIOCTLCMD &&
+#  endif
+		errno != EINVAL) return -1;
 #endif
 
 #ifdef __linux__
