Index: git/tests/ts_calibrate.c
===================================================================
--- git.orig/tests/ts_calibrate.c
+++ git/tests/ts_calibrate.c
@@ -256,7 +256,7 @@ int main()
 			cal_fd = open (TS_POINTERCAL, O_CREAT | O_RDWR,
 			               S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
 		}
-		len = sprintf(cal_buffer,"%d %d %d %d %d %d %d %d %d",
+		len = sprintf(cal_buffer,"%d %d %d %d %d %d %d %d %d\n",
 		              cal.a[1], cal.a[2], cal.a[0],
 		              cal.a[4], cal.a[5], cal.a[3], cal.a[6],
 		              xres, yres);
