Index: lib/bb/cooker.py
===================================================================
--- lib/bb/cooker.py.orig
+++ lib/bb/cooker.py
@@ -919,11 +919,6 @@ class BBCooker:
         return (finalfiles, masked)
 
     def serve(self):
-
-        # Empty the environment. The environment will be populated as
-        # necessary from the data store.
-        bb.utils.empty_environment()
-
         if self.configuration.profile:
             try:
                 import cProfile as profile
