Index: bitbake/lib/bb/parse/parse_py/BBHandler.py
===================================================================
--- lib/bb/parse/parse_py/BBHandler.py
+++ lib/bb/parse/parse_py/BBHandler.py
@@ -146,7 +146,11 @@ def handle(fn, d, include):
     if ext != ".bbclass":
         data.setVar('FILE', fn, d)
 
-    statements.eval(d)
+    try:
+        statements.eval(d)
+    except bb.parse.SkipPackage:
+        bb.data.setVar("__SKIPPED", True, d)
+        return { "" : d }
 
     if ext == ".bbclass":
         classes.remove(__classname__)
