Index: lib/bb/build.py
===================================================================
--- lib/bb/build.py.orig
+++ lib/bb/build.py
@@ -276,10 +276,11 @@ def exec_task(task, d):
     if not data.getVarFlag(task, 'task', d):
         raise EventException("No such task", InvalidTask(task, d))
 
+    localdata = data.createCopy(d)
+
     try:
         bb.msg.debug(1, bb.msg.domain.Build, "Executing task %s" % task)
-        old_overrides = data.getVar('OVERRIDES', d, 0)
-        localdata = data.createCopy(d)
+        old_overrides = data.getVar('OVERRIDES', localdata, 0)
         data.setVar('OVERRIDES', 'task-%s:%s' % (task[3:], old_overrides), localdata)
         data.update_data(localdata)
         data.expandKeys(localdata)
@@ -294,8 +295,8 @@ def exec_task(task, d):
             logfile = None
             msg = message
         bb.msg.note(1, bb.msg.domain.Build, "Task failed: %s" % message )
-        failedevent = TaskFailed(msg, logfile, task, d)
-        event.fire(failedevent, d)
+        failedevent = TaskFailed(msg, logfile, task, localdata)
+        event.fire(failedevent, localdata)
         raise EventException("Function failed in task: %s" % message, failedevent)
 
     # make stamp, or cause event and raise exception
