Index: bitbake/lib/bb/cooker.py
===================================================================
--- lib/bb/cooker.py
+++ lib/bb/cooker.py
@@ -149,9 +149,8 @@ class BBCooker:
             self.commandlineAction = ["showVersions"]
         elif self.configuration.parse_only:
             self.commandlineAction = ["parseFiles"]
-        # FIXME - implement
-        #elif self.configuration.interactive:
-        #    self.interactiveMode()
+        elif self.configuration.interactive:
+            self.interactiveMode()
         elif self.configuration.dot_graph:
             if self.configuration.pkgs_to_build:
                 self.commandlineAction = ["generateDotGraph", self.configuration.pkgs_to_build, self.configuration.cmd]
