Index: ui/knotty.py
===================================================================
--- lib/bb/ui.orig/knotty.py
+++ lib/bb/ui/knotty.py
@@ -84,7 +84,6 @@ def init(server, eventHandler):
                 print 'WARNING: ' + event._message
                 continue
             if isinstance(event, bb.msg.MsgError):
-                return_value = 1
                 print 'ERROR: ' + event._message
                 continue
             if isinstance(event, bb.msg.MsgFatal):
