Index: bitbake/lib/bb/taskdata.py
===================================================================
--- lib/bb/taskdata.py
+++ lib/bb/taskdata.py
@@ -34,7 +34,7 @@ def re_match_strings(target, strings):
 
     for name in strings:
         if (name==target or
-                re.search(name,target)!=None):
+                re.match(name,target)!=None):
             return True
     return False
 
